### PR TITLE
Query Browser: Add query autocompletion and split query browser code into separate files

### PR DESCRIPTION
**Pull request by kyoto**
The first commit adds autocompletion to the query input. PromQL functions and available metrics are suggested as you type.

The second commit is just moving Query Browser related components and CSS into new files in `components/monitoring/`.

autocomplete

### CODE REVIEW

File: frontend/public/components/monitoring.tsx [hunk 1]
```diff
@@ -1068,21 +1122,85 @@ const SeriesButton = ({colorIndex, isDisabled, onClick}) => {
...
+    setToken('');
+    inputRef.current.focus();
+  };
+
+  const functionOptions = token.length < 2 ? [] : prometheusFunctions.filter(a => fuzzy(token, a));
+  const metricOptions = token.length < 2 ? [] : metrics.filter(a => fuzzy(token, a));
+
+  return <div className="query-browser__query query-browser__inline-control pf-c-dropdown">
+    <textarea
+      autoFocus={true}
```
Comment (spadgett) [inline frontend/public/components/monitoring.tsx]: nit: not new, but prefer

```tsx
<textarea
  autoFocus
```

without `={true}`

File: frontend/public/components/monitoring.tsx [hunk 2]
```diff
@@ -1068,21 +1122,85 @@ const SeriesButton = ({colorIndex, isDisabled, onClick}) => {
...
+      onKeyDown={onKeyDown}
+      placeholder="Expression (press Shift+Enter for newlines)"
+      ref={inputRef}
+      value={value}
+    />
+    {(!_.isEmpty(functionOptions) || !_.isEmpty(metricOptions)) && (
+      <ul className="pf-c-dropdown__menu query-browser__metrics-dropdown-menu">
+        {!_.isEmpty(functionOptions) && <div className="text-muted query-browser__dropdown--subtitle">Functions</div>}
+        {_.map(functionOptions, f => <li key={f}>
+          <a href="#" className="pf-c-dropdown__menu-item" data-autocomplete={f} onClick={onClick}>{f}</a>
```
Comment (spadgett) [inline frontend/public/components/monitoring.tsx]: nit: for accessibility, these would be better as `<button className="btn btn-link" type="button">`, but I'm not sure if that breaks the patternfly styles

File: frontend/public/components/monitoring.tsx (hunk 1 above)
Comment (kyoto) [reply frontend/public/components/monitoring.tsx]: Fixed by https://github.com/openshift/console/pull/1896

File: frontend/public/components/monitoring.tsx (hunk 2 above)
Comment (kyoto) [reply frontend/public/components/monitoring.tsx]: PR: https://github.com/openshift/console/pull/1996 (but without `btn btn-link` to get the correct styling).

**Pull request merged**